src/cluster_legacy.c

```diff
@@ -2594,6 +2672,36 @@ uint32_t writePingExt(clusterMsg *hdr, int gossipcount)  {
     totlen += getShardIdPingExtSize();
     extensions++;
 
+    /* Populate insternal secret */
+    if (cursor != NULL) {
+        clusterMsgPingExtInternalSecret *ext = preparePingExt(cursor, CLUSTERMSG_EXT_TYPE_INTERNALSECRET, getInternalSecretPingExtSize());
+        memcpy(ext->internal_secret, myself->internal_secret, CLUSTER_INTERNALSECRETLEN);
+
+        /* Move the write cursor */
+        cursor = nextPingExt(cursor);
+    }
+    totlen += getInternalSecretPingExtSize();
+    extensions++;
+
+    for (int i = 0 ; i < gossipcount ; ++i) {
```

Comment:
@ShooterIT lets continue this discussion here: https://github.com/redis/redis/pull/13740#discussion_r1930255241

Reply:
hi @MeirShpilraien  why might it take longer to converge? Although a node only gossip its own secret, this secret is the smallest one found among all the communicating nodes. every node will update their secret to the smallest one.

after a node restart, maybe only one gossip ping message can let the node have the correct smallest secret

Reply:
OK I am convinced, I will change it. Thanks.

Reply:
Fixed, thanks.